src/pages/api/revalidate-all-tutorials.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { NextApiRequest, NextApiResponse } from 'next/types'
+import { StatusCodes } from 'http-status-codes'
+
+import { validateToken } from 'lib/api-validate-token'
+import { getAllCollections } from 'lib/learn-client/api/collection'
+import {
+	Collection as ClientCollection,
+	TutorialLite as ClientTutorialLite,
+} from 'lib/learn-client/types'
+import { activeProductSlugs } from 'lib/products'
+import { SectionOption } from 'lib/learn-client/types'
+import { getTutorialSlug } from 'views/collection-view/helpers'
+import { getCollectionSlug } from 'views/collection-view/helpers'
+import { ProductSlug } from 'types/products'
+
+const BATCH_SIZE = 10
+const DELAY_TIME = 200 // ms
+
+/**
+ * Accepts a POST request, triggers revalidation for all tutorial paths for all products
+ * landing pages, collection pages, and tutorial pages associated
+ */
+async function handler(request: NextApiRequest, response: NextApiResponse) {
+	if (request.method !== 'POST') {
+		response.status(StatusCodes.NOT_FOUND)
+		return
+	}
+
+	try {
+		const tutorialLandingPaths = getTutorialLandingPaths()
+		const collectionAndTutorialPaths = await getCollectionAndTutorialPaths()
+		const paths: string[] = [
+			...tutorialLandingPaths,
+			...collectionAndTutorialPaths,
+		]
+
+		if (!paths || paths.length === 0) {
+			response.status(200).end()
+		}
+
+		// Loop over all paths to revalidate in batches
+		// as this endpoint will fire off >1000 revalidation requests
+		let batchRevalidatePromises = []
+		for (let i = 0; i < paths.length; i++) {
+			// remove any trailing slash
+			const path = paths[i].replace(/\/$/, '')
+			console.log('[Revalidate]', path)
+			batchRevalidatePromises.push(response.revalidate(path))
+
+			// flush the batch every N paths, or at the end of the loop
+			if (
+				batchRevalidatePromises.length >= BATCH_SIZE ||
+				i >= paths.length - 1
+			) {
+				await Promise.allSettled(batchRevalidatePromises)
+				// delay to ensure we don't overwhelm the server
+				await delay(DELAY_TIME)
```

Comment:
Wanted to keep this simple and but open to suggestions here! I saw issues when only doing the batching (this is like ~1800 paths? according to my tests) and the delay makes things run great locally.

Reply:
FWIW, I don't think we have run into any rate limiting with the docs revalidation endpoint yet, and a few of the products have >500 page that get revalidated in one go without any throttling (locally, it's a no-op so that shouldn't be a concern).

We do have `p-map` in our dependencies already, so that could be a good solution to add some throttling!

Reply:
Okay, I'll remove the delay for now and watch after we roll it out!

Reply:
I think p-map only supports concurrency limits but not throttling? I added a note reference both p-map and p-throttle in case we need to add this later on